test/UID2.Client.Test/EncryptionTestsV3.cs

```diff
@@ -20,19 +20,40 @@ public class EncryptionTestsV3
         private static readonly string EXAMPLE_UID = "ywsvDNINiZOVSsfkHpLpSJzXzhr6Jx9Z/4Q0+lsEUvM=";
         private static readonly string CLIENT_SECRET = "ioG3wKxAokmp+rERx6A4kM/13qhyolUXIu14WN16Spo=";
 
+        [Theory]
+        [InlineData("A", nameof(IdentityScope.UID2), IdentityType.Email)]
+        [InlineData("B", nameof(IdentityScope.UID2), IdentityType.Phone)]
+        [InlineData("E", nameof(IdentityScope.EUID), IdentityType.Email)]
+        [InlineData("F", nameof(IdentityScope.EUID), IdentityType.Phone)]
+        public void IdentityScopeAndType_TestCases(String idType, string identityScope, IdentityType? identityType)
+        {
+            var client = new UID2Client("ep", "ak", CLIENT_SECRET, Enum.Parse<IdentityScope>(identityScope));
+            var refreshResult = client.RefreshJson(KeySetToJson(MASTER_KEY, SITE_KEY));
+            Assert.True(refreshResult.Success);
+            
+            string advertisingToken = idType + UID2TokenGenerator.GenerateUid2TokenV3(EXAMPLE_UID, MASTER_KEY, SITE_ID, SITE_KEY, UID2TokenGenerator.DefaultParams).Substring(1);
```

Comment:
rather than manipulating a token with different prefixes, can we instead use pre-generated tokens?

Reply:
you need to create a v3 raw UID2 (not v3 token) which would have the right identityType info which can then be used in GenerateUid2TokenV3/V4 for token's IdentityType info. The v2 raw UID doesn't have identity type info (see operator repo's com.uid2.operator.service.TokenUtils#getAdvertisingIdV2). And prod UID is using v2 raw UID actually and we won't change in foreseeable future. 

If you go down the path to generate a v3 raw UID, then you can call GenerateUid2TokenV3/V4 to generate realistic ad token dynamically. You can either generate token dynamically like above, or hardcode these pre-generated tokens as static readonly string crossPlatform and see how in ConsistencyCheck_Decrypt we change the master/site key expiry to effectively infinite so unit tests always can run and decrypt for next 20 years lol